src/server/v2.0/handler/robot.go

```diff
@@ -293,6 +270,60 @@ func (rAPI *robotAPI) validate(d int64, level string, permissions []*models.Perm
 	return nil
 }
 
+// only disable can be updated for v1 robot
+func (rAPI *robotAPI) updateV1Robot(ctx context.Context, params operation.UpdateRobotParams, r *robot.Robot) error {
+	if err := rAPI.requireAccess(ctx, params.Robot.Level, r.ProjectID, rbac.ActionUpdate); err != nil {
+		return err
+	}
+	r.Disabled = params.Robot.Disable
+	r.Description = params.Robot.Description
+	if err := rAPI.robotCtl.Update(ctx, r, &robot.Option{
+		WithPermission: false,
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (rAPI *robotAPI) updateV2Robot(ctx context.Context, params operation.UpdateRobotParams, r *robot.Robot) error {
+	if err := rAPI.validate(params.Robot.Duration, params.Robot.Level, params.Robot.Permissions); err != nil {
+		return err
+	}
+
+	if err := rAPI.requireAccess(ctx, params.Robot.Level, params.Robot.Permissions[0].Namespace, rbac.ActionUpdate); err != nil {
+		return err
+	}
+
+	if params.Robot.Level != r.Level || params.Robot.Name != r.Name {
+		return errors.BadRequestError(nil).WithMessage("cannot update the level or name of robot")
+	}
+
+	if r.Duration != params.Robot.Duration {
+		r.Duration = params.Robot.Duration
+		if params.Robot.Duration == -1 {
+			r.ExpiresAt = -1
+		} else if params.Robot.Duration == 0 {
+			r.Duration = int64(config.RobotTokenDuration())
+			r.ExpiresAt = r.CreationTime.AddDate(0, 0, config.RobotTokenDuration()).Unix()
```

Comment:
The default value for the RobotTokenDuration is 43200 minutes, AddDate() add by days, so it could be an error.

Reply:
Need also verify the default expires in the creating of robot account.

Reply:
the default RobotTokenDuration is in days, not in minutes.

Reply:
https://github.com/goharbor/harbor/blob/60e3668d433315f82874aacd2fbf5d84f3f45474/src/common/config/metadata/metadatalist.go#L152-L151

Reply:
resolved